server/channels/db/migrations/mysql/000117_msteams_shared_channels.up.sql

```diff
@@ -0,0 +1,72 @@
+SET @preparedStatement = (SELECT IF(
+    NOT EXISTS(
+        SELECT 1 FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'RemoteClusters'
+        AND table_schema = DATABASE()
+        AND column_name = 'PluginID'
+    ),
+    'ALTER TABLE RemoteClusters ADD COLUMN PluginID varchar(190) NOT NULL DEFAULT \'\';',
+    'SELECT 1;'
+));
+
+PREPARE addColumnIfNotExists FROM @preparedStatement;
+EXECUTE addColumnIfNotExists;
+DEALLOCATE PREPARE addColumnIfNotExists;
+
+
+SET @preparedStatement = (SELECT IF(
+    NOT EXISTS(
+        SELECT 1 FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'SharedChannelRemotes'
+        AND table_schema = DATABASE()
+        AND column_name = 'LastPostCreateAt'
+    ),
+    'ALTER TABLE SharedChannelRemotes ADD COLUMN LastPostCreateAt bigint NOT NULL DEFAULT 0;',
+    'SELECT 1;'
+));
+
+PREPARE addColumnIfNotExists FROM @preparedStatement;
+EXECUTE addColumnIfNotExists;
+DEALLOCATE PREPARE addColumnIfNotExists;
+
+
+SET @preparedStatement = (SELECT IF(
+    NOT EXISTS(
+        SELECT 1 FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'SharedChannelRemotes'
+        AND table_schema = DATABASE()
+        AND column_name = 'LastPostCreateID'
+    ),
+    'ALTER TABLE SharedChannelRemotes ADD COLUMN LastPostCreateID varchar(26);',
+    'SELECT 1;'
+));
+
+PREPARE addColumnIfNotExists FROM @preparedStatement;
+EXECUTE addColumnIfNotExists;
+DEALLOCATE PREPARE addColumnIfNotExists;
+
+-- rename `lastpostid` column only if it exists and the new name doesn't exist.
```

Comment:
Is this a breaking schema change, in that peers in the HA cluster that upgrade incrementally will immediately break?

Reply:
Yes indeed. Good point. I think I had this in my doc somewhere, but yeah we cannot rename the column atomically any more. Our schema migration guidelines guarantee that old nodes in an HA cluster will still work correctly as the new node gets migrated.

IF we want to rename this, we need to:
1. Create the new column with the new name.
2. Start a batch migration worker which will slowly migrate values from the old column to the new, and at the same time, do a double write from within the code.
3. In the next ESR, drop the old column, and start using only the new column.

Long story short, it's a long winded process now. If it's not too much of a problem, I might suggest to not do the rename altogether.

Reply:
Rename of column has been reverted.